feat: https redirect with avosapps.us

weakish opened the pull request.

weakish committed:
feat: https redirect with avosapps.us

leeyeh commented:
为什么要判断域名呢？跳过 localhost?

weakish commented:
对哦，直接跳过 localhost 应该就可以（或者说判断 host 是一个 FQDN  就可以）。这里以前这么写是因为很久以前云引擎不支持绑域名或者绝大多数云引擎没有绑自己的域名？  @jysperm

leeyeh commented:
`AV.Cloud.__prod` 这个条件也很奇怪。

jysperm commented:
https://github.com/leancloud/leanengine-node-sdk/commit/ef20d26fe9536ba8bf60ed4e66102e71fc8f6f85
https://github.com/leancloud/leanengine-node-sdk/issues/44

weakish committed:
fix: tweak https redirect condition

1. Remove domain name detection since applications should use custom domain.
2. Bypass cloud function health check.
3. Test X-Forwarded-Proto header (no redirect on `lean up`).

jysperm commented:
https://github.com/leancloud/leanengine-node-sdk/pull/150
